code-challenges.test.js

```diff
@@ -1,105 +1,188 @@
+  }
+  return newArray
+}
+
+// console.log(shuffleArray(colors1));
+// console.log(shuffleArray(colors2));
+//
+//
+//
+// // --------------------2) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
+//
+// // a) Create a test with expect statements for each of the variables provided.
+//
 var cubeAndSum1 = [2, 3, 4]
 // Expected output: 99
 var cubeAndSum2 = [0, 5, 10]
 // Expected output: 1125
 
-
-
-
-
-// b) Create the function that makes the test pass.
-
-
-
-
-
-// --------------------3) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
-
-// a) Create a test with expect statements for each of the variables provided.
-
+describe("When 'cubeAndSumArray' is called", () => {
+  it("return the sum of the numbers in array cubed", () => {
+    expect(cubeAndSumArray(cubeAndSum1)).toEqual(99)
+    expect(cubeAndSumArray(cubeAndSum2)).toEqual(1125)
+  })
+})
+//
+//
+//
+// // b) Create the function that makes the test pass.
+// //create a custom function that takes in an array as an arguement
+// //cube each number using foreach
+// //use reduce to iterate over array and find the sum of the numbers in array.
+//
+const cubeAndSumArray = (array) => {
+  let cubedArray = []
+  array.forEach((value,index) => {
+    cubedArray.push(value ** 3)
+  })
+  return cubedArray.reduce((a, b) => a + b, 0)
+}
+
+console.log(cubeAndSumArray(cubeAndSum1));
+console.log(cubeAndSumArray(cubeAndSum2));
+// //
+// //
+// // // --------------------3) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
+// //
+// // // a) Create a test with expect statements for each of the variables provided.
+// //
 var nums1 = [3, 56, 90, -8, 0, 23, 6]
 // Expected output: [-8, 90]
 var nums2 = [109, 5, 9, -59, 8, 24]
 // Expected output: [-59, 109]
 
-
-
-
-
-// b) Create the function that makes the test pass.
-
-
-
-
-
-// --------------------4) Create a function that takes in a string and returns a string with every other letter capitalized.
-
-// a) Create a test with expect statements for each of the variables provided.
-
+describe("When 'onlyMinMax' is called", () => {
+  it("returns with only 2 numbers in array that are smallest and largest", () => {
+    expect(onlyMinMax(nums1)).toEqual([-8, 90])
+    expect(onlyMinMax(nums2)).toEqual([-59, 109])
+  })
+})
+// //
+// // // b) Create the function that makes the test pass.
+// // //Create a custom function that takes an array of numbers as an arguement
+// // //use filter to find smallest number and largest number in array. -- refactored this as I could not get filter to show both min num and max num.
+// // // refactor shows finding the minimum and maximum numbers by using math function in a variable and pushing into an empty array.
+// //
+const onlyMinMax = (array) => {
+  let minMaxArray = []
+  let minNum = Math.min(...array)
```

Comment:
good logic. However, consider a way of doing this that does not require multiple iterations on the given array.